### PR TITLE
Fix technical documentation links 20240531

### PR DESCRIPTION
**Pull request by Sergiodero**
This PR fixes multiple broken links in the Technical documentation section. This is based on the following list of warnings:

> - WARNING -  Doc file 'documentation/overview.md' contains a link '(./Realtime/reference.md)', but the target 'documentation/(./Realtime/reference.md)' is not found among documentation files.
> - WARNING -  Doc file 'documentation/Realtime/proto.md' contains a link 'gtfs-realtime.proto', but the target 'documentation/Realtime/gtfs-realtime.proto' is not found among documentation files.
> - WARNING -  Doc file 'documentation/Realtime/reference.md' contains a link 'trip-updates.md', but the target 'documentation/Realtime/trip-updates.md' is not found among documentation files.
> - WARNING -  Doc file 'documentation/Realtime/reference.md' contains a link 'trip-updates.md', but the target 'documentation/Realtime/trip-updates.md' is not found among documentation files.
> - WARNING -  Doc file 'documentation/Realtime/reference.md' contains a link 'trip-modifications.md', but the target 'documentation/Realtime/trip-modifications.md' is not found among documentation files.
> - WARNING -  Doc file 'documentation/schedule/reference.md' contains a link 'examples/2-leg.svg', but the target 'documentation/schedule/examples/2-leg.svg' is not found among documentation files.
> - WARNING -  Doc file 'documentation/schedule/reference.md' contains a link 'examples/3-leg.svg', but the target 'documentation/schedule/examples/3-leg.svg' is not found among documentation files.
> - WARNING -  Doc file 'documentation/schedule/examples/continuous-stops.md' contains a link '../../assets/victor-valley-transit.svg', but the target 'documentation/assets/victor-valley-transit.svg' is not found among documentation files.
> - WARNING -  Doc file 'documentation/schedule/examples/flex.md' contains a link '../../../assets/on-demand_services_between_multiple_zones(r).zip', but the target 'assets/on-demand_services_between_multiple_zones(r).zip' is not found among documentation files.
> - WARNING -  Doc file 'documentation/schedule/examples/flex.md' contains a link '../../../assets/deviated_drop-off_route.zip', but the target 'assets/deviated_drop-off_route.zip' is not found among documentation files.

@isabelle-dr @tzujenchanmbd do you mind having a look?

### CODE REVIEW

File: docs/documentation/overview.md
Comment (eliasmbd) [inline docs/documentation/overview.md:1]: Shouldn't we mark links without the .md file extension? I know that it is a best practice for the links and sometimes if you leave the extension it doesn't work.

**Review by eliasmbd — APPROVED** [review]
This is my only concern if not everything else LGTM

File: docs/documentation/overview.md
Comment (Sergiodero) [reply docs/documentation/overview.md:1]: Right, we have two options, we either use full links with no extension (e.g.**/documentation/Realtime/reference**), or short links with extension (e.g. **../Realtime/reference.md**). I don't have a lot of clarity on this, but I think the shor-link with extension option might be helpful if we're moving complete sections in the site structure. Having said that I don't think we'll be doing this much in the near future.

Comment (Sergiodero) [reply docs/documentation/overview.md:1]: @fredericsimard do you have any thoughts on which option might be a better best practice?

Comment (fredericsimard) [reply docs/documentation/overview.md:1]: At this stage (that generates these errors), it's looking and expecting a file ending with .md and not a URL like on a website. AFAIK.

Comment (eliasmbd) [reply docs/documentation/overview.md:1]: So under this structure i've had links not work ../Realtime/reference.md, I have to remove the .md here. HEre is how i did the link on my recent work. 
image

Comment (fredericsimard) [reply docs/documentation/overview.md:1]: It's possible, normally if you're referring to a file that exists within the repo, it should end with the extension.

**Comment by Sergiodero** [comment]
Thanks @eliasmbd I harmonized all links in the Technical Doc Overview page. I'm merging after this last commit.

**Pull request merged**